Typo fixes (#556)
rust-lang-nursery/reference: src/type-coercions.md

anywhere else a coercion can occur.

Two traits, [`Unsize`] and [`CoerceUnsized`], are used
to assist in this process and expose it for library use. The following

> coercions are compiler built-ins and, if `T` can be coerced to `U` with one of them, then

Reviewer: ```suggestion
coercions are built-ins and, if `T` can be coerced to `U` with one of them, then
```

> the compiler will provide an implementation of `Unsize<U>` for `T`:

Reviewer: ```suggestion
an implementation of `Unsize<U>` for `T` will be provided:
```
